File: src/main/java/com/clearcont/clearcontapp/repository/CustomerContabilRepository.java
package com.clearcont.clearcontapp.repository;

import com.clearcont.clearcontapp.model.CustomerContabil;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CustomerContabilRepository extends JpaRepository<CustomerContabil, Long> {
    @Transactional
    List<CustomerContabil> findAllByComposicaoLancamentosContabeis_Balancete_Id(Long id);
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/repository/CustomerContabilRepository.kt
package br.com.clearcont.clearcontwebapp.repository

import br.com.clearcont.clearcontwebapp.models.CustomerContabil
import jakarta.transaction.Transactional
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface CustomerContabilRepository : JpaRepository<CustomerContabil, Long> {
    @Transactional
    @Query("SELECT c FROM CustomerContabil c WHERE c.composicaoLancamentosContabeis.balancete.id = ?1")
    fun findCustomerContabilByBalanceteId(id: Long): List<CustomerContabil>
}
